crates/trie/parallel/src/proof_task.rs

```diff
@@ -258,15 +263,21 @@ fn storage_worker_loop<Tx>(
 /// # Shutdown
 ///
 /// Worker shuts down when the crossbeam channel closes (all senders dropped).
-fn account_worker_loop<Tx>(
-    proof_tx: ProofTaskTx<Tx>,
+fn account_worker_loop<Factory>(
+    view: ConsistentDbView<Factory>,
+    task_ctx: ProofTaskCtx,
     work_rx: CrossbeamReceiver<AccountWorkerJob>,
     storage_work_tx: CrossbeamSender<StorageWorkerJob>,
     worker_id: usize,
     #[cfg(feature = "metrics")] metrics: ProofTaskTrieMetrics,
 ) where
-    Tx: DbTx,
+    Factory: DatabaseProviderFactory<Provider: BlockReader>,
 {
+    // Create db transaction before entering work loop
+    let provider = view.provider_ro()
+        .expect("Account worker failed to initialize: database unavailable");
```

Comment:
instead of having expect 

https://github.com/paradigmxyz/reth/pull/19012/files#diff-e3fb5658ad9c8b5024a6858cc898487c1961023c6361aeab5b4dcdd7ef781fb6L1231-L1233

its much simpler to expect here, and we dont have to propogate the error up